lib/src/libp2p/with_buffers.rs

```diff
@@ -33,12 +33,12 @@ use core::{
     task::Poll,
 };
 use futures_util::{AsyncRead, AsyncWrite};
-use std::{io, time::Instant};
+use std::io;
 
 /// Holds an implementation of `AsyncRead` and `AsyncWrite`, alongside with a read buffer and a
 /// write buffer.
 #[pin_project::pin_project]
-pub struct WithBuffers<T> {
+pub struct WithBuffers<T, TNow> {
```

Comment:
Could you change `T` to `TSocket` everywhere, since there are two parameters now?

Reply:
Yep that makes sense thanks!